# Sequencing Data {#sequencing-data}

Sequencing data may include:

1. Research projects
2. From Sequence Read Archive (SRA)
3. In `silico` simulation


## From research projects
The most common sources of sequencing data are from research projects. There are multiple sequencing platforms out there. For example, Illumina sequencing company uses technologies capable of profiling entire microbial communities present in environmental. Example include: 

- 16S rRNA Sequencing
- Shotgun Metagenomic Sequencing

## From archives such as NCBI-SRA
The NCBI Sequence Read Archive (SRA) stores sequencing data from the next generation sequencing platforms. Users can download data from the SRA archive. Below are minimal steps to do that:

### Install SRA Toolkit on Mac OS.
- Navigate to where you want to install the tools, preferably the home directory.
- For more information click [here](https://github.com/ncbi/sra-tools/wiki/02.-Installing-SRA-Toolkit).

```{}
curl -LO  https://ftp-trace.ncbi.nlm.nih.gov/sra/sdk/3.0.0/sratoolkit.3.0.0-mac64.tar.gz
tar -xf sratoolkit.3.0.0-mac64.tar.gz

export PATH=$HOME/sratoolkit.3.0.0-mac64/bin/:$PATH

```

### Create a cache root directory
```{}
mkdir -p ~/ncbi
echo '/repository/user/main/public/root = "/Volumes/SeagateTMB/SRA/BUSHMEAT/SRR7450"' > ~/ncbi/user-settings.mkfg
```


### Confirm sra toolkit configuration
- Command below will display a blue colored dialog.
- Use tab or click `c` to navigate to cache tab.
- Review the configuration then save `s` and exit `x`.

```{}
vdb-config -i
```


![A screenshot of the SRA configuration.](images/sra_config_cache.png){width=100%}

For more information click [here](https://github.com/ncbi/sra-tools/wiki/03.-Quick-Toolkit-Configuration).

<br>

### Start downloading the fastq files
- Be sure that the `fasterq-dump` is in the path.
- Type `which fasterq-dump` to confirm if is in the path.
- Use the absolute path like `~/sratoolkit.3.0.0-mac64/bin/fasterq-dump` if not already exported to the path.
- The example below uses absolute path to execute the `fasterq-dump` command.

> Notice that the output and temporary files are placed in an external drive mounted to the computer.

> The sratoolkit folder is in the home directory.

```{}
for (( i = 706; i <= 761; i++ ))
	do
		time ~/sratoolkit.3.0.0-mac64/bin/fasterq-dump SRR7450$i \
		-O /Volumes/SeagateTMB/SRA/BUSHMEAT/SRR7450/ \
		-t /Volumes/SeagateTMB/SRA/BUSHMEAT/SRR7450/tmpfiles \
		--threads 4 \
		--split-3;
	done
```


## Simmulated insilico sequencing data
- Most insilico data is used for testing software before applying real data.
- How to generate insilico data can be challenging. 
- Below is a demo for generating NovaSeq, MiSeq and HiSeq insilico sequencing data.

### Demo1: NovaSeq metagenomics data

- Five novaseq paired fastq files

```{}
iss generate \
	--ncbi bacteria viruses archaea -U 15 0 2 \
	--coverage lognormal \
	--n_reads 0.12M \
	--model novaseq \annual report
	--output data/novaseq_01 \
	--compress \
	--seed 1
iss generate \
	--ncbi bacteria viruses archaea -U 15 1 1 \
	--coverage lognormal \
	--n_reads 0.11M \
	--model novaseq \
	--output data/novaseq_02 \
	--compress \
	--seed 2
iss generate \
	--ncbi bacteria viruses archaea -U 10 0 0 \
	--coverage lognormal \
	--n_reads 0.1M \
	--model novaseq \
	--output data/novaseq_03 \
	--compress \
	--seed 3
iss generate \
	--ncbi bacteria viruses archaea -U 10 0 0 \
	--coverage lognormal \
	--n_reads 0.1M \
	--model novaseq \
	--output data/novaseq_04 \
	--compress \
	--seed 4
iss generate \
	--ncbi bacteria viruses archaea -U 10 0 0 \
	--coverage lognormal \
	--n_reads 0.1M \
	--model novaseq \
	--output data/novaseq_05 \
	--compress \
	--seed 5
	
```	

### Demo2: MiSeq  metagenomics data
- Five miseq paired fastq files

```{}
iss generate \
	--ncbi bacteria viruses archaea -U 10 0 2 \
	--coverage lognormal \
	--n_reads 0.12M \
	--model miseq \
	--output data/miseq_01 \
	--compress \
	--seed 16
iss generate \
	--ncbi bacteria viruses archaea -U 15 1 1 \
	--coverage lognormal \
	--n_reads 0.11M \
	--model miseq \
	--output data/miseq_02 \
	--compress \
	--seed 17
iss generate \
	--ncbi bacteria viruses archaea -U 5 0 0 \
	--coverage lognormal \
	--n_reads 0.1M \
	--model miseq \
	--output data/miseq_03 \
	--compress \
	--seed 18
iss generate \
	--ncbi bacteria viruses archaea -U 15 0 0 \
	--coverage lognormal \
	--n_reads 0.1M \
	--model miseq \
	--output data/miseq_04 \
	--compress \
	--seed 19
iss generate \
	--ncbi bacteria viruses archaea -U 5 0 0 \
	--coverage lognormal \
	--n_reads 0.1M \
	--model miseq \
	--output data/miseq_05 \
	--compress \
	--seed 20
	
```	
	
### Demo3: HiSeq  metagenomics data
- Five hiseq paired fastq files

```{}
iss generate \
	--ncbi bacteria viruses archaea -U 10 1 2 \
	--coverage lognormal \
	--n_reads 0.12M \
	--model hiseq \
	--output data/hiseq_01 \
	--compress \
	--seed 21
iss generate \
	--ncbi bacteria viruses archaea -U 15 1 1 \
	--coverage lognormal \
	--n_reads 0.11M \
	--model hiseq \
	--output data/hiseq_02 \
	--compress \
	--seed 22
iss generate \
	--ncbi bacteria viruses archaea -U 10 0 0 \
	--coverage lognormal \
	--n_reads 0.1M \
	--model hiseq \
	--output data/hiseq_03 \
	--compress \
	--seed 23
iss generate \
	--ncbi bacteria viruses archaea -U 10 0 0 \
	--coverage lognormal \
	--n_reads 0.1M \
	--model hiseq \
	--output data/hiseq_04 \
	--compress \
	--seed 24
iss generate \
	--ncbi bacteria viruses archaea -U 10 0 0 \
	--coverage lognormal \
	--n_reads 0.1M \
	--model hiseq \
	--output data/hiseq_05 \
	--compress \
	--seed 25

```


<br>

## How to resize fastq files 
- Sometimes we want to extract a small subset to test the bioinformatics pipeline.
- You can resize the fastq files using a `seqkit sample` function [@Shen2016; @seqkit2022].
- Below is a demo for extracting only 0.1% of the sequencing data.

```{}
mkdir -p data
for i in {706..761}
  do
    cat SRR7450$i\_R1.fastq \
    | seqkit sample -p 0.01 \
    | seqkit shuffle -o data/SRR7450$i\_R1_sub.fastq \
    | cat SRR7450$i\_R2.fastq \
    | seqkit sample -p 0.01 \
    | seqkit shuffle -o data/SRR7450$i\_R2_sub.fastq
  done

```




