# Sequencing Data {#raw-data}

Read sequencing data may be obtained from different sources. The most common ones include:

1. Reads from sequencing platforms for research purposes.
2. Reads downloaded from the Sequence Read Archive (SRA).
3. Reads synthesized `in-silico` using special simulation software


## Reads from sequencing platforms
The most common sources of sequencing data are from research projects. There are multiple sequencing platforms out there. For example, Illumina sequencing company uses technologies capable of profiling entire microbial communities present in environmental. Example include: 

- 16S rRNA Sequencing
- Shotgun Metagenomic Sequencing

## Reads from archives such as NCBI-SRA
The NCBI Sequence Read Archive (SRA) stores sequencing data from the next generation sequencing platforms. Users can download data from the SRA archive. Below are minimal steps to do that:

### Install SRA Toolkit on Mac OS.
- Navigate to where you want to install the tools, preferably the home directory.
- For more information click [here](https://github.com/ncbi/sra-tools/wiki/02.-Installing-SRA-Toolkit).

```{}
curl -LO  https://ftp-trace.ncbi.nlm.nih.gov/sra/sdk/3.0.0/sratoolkit.3.0.0-mac64.tar.gz
tar -xf sratoolkit.3.0.0-mac64.tar.gz

export PATH=$HOME/sratoolkit.3.0.0-mac64/bin/:$PATH

```

### Create a cache root directory
```{}
mkdir -p ~/ncbi
echo '/repository/user/main/public/root = "/Volumes/SeagateTMB/SRA/BUSHMEAT/SRR7450"' > ~/ncbi/user-settings.mkfg
```


### Confirm sra toolkit configuration
- Command below will display a blue colored dialog.
- Use tab or click `c` to navigate to cache tab.
- Review the configuration then save `s` and exit `x`.

```{}
vdb-config -i
```


![A screenshot of the SRA configuration.](images/sra_config_cache.png){width=100%}

<br>

For more information click [here](https://github.com/ncbi/sra-tools/wiki/03.-Quick-Toolkit-Configuration).

### Downloading the fastq files
- Be sure that the `fasterq-dump` is in the path.
- Type `which fasterq-dump` to confirm if is in the path.
- Use the absolute path like `~/sratoolkit.3.0.0-mac64/bin/fasterq-dump` if the executable file is not in the path.
- The example below uses absolute path to execute the `fasterq-dump` command.

> Notice that the output and temporary files are placed in an external drive mounted to the computer.

> The sratoolkit folder is in the home directory.

```{}
for (( i = 706; i <= 761; i++ ))
	do
		time ~/sratoolkit.3.0.0-mac64/bin/fasterq-dump SRR7450$i \
		-O /Volumes/SeagateTMB/SRA/BUSHMEAT/SRR7450/ \
		-t /Volumes/SeagateTMB/SRA/BUSHMEAT/SRR7450/tmpfiles \
		--threads 4 \
		--split-3;
	done
```


## Synthetic metagenomics Reads {#synthetic-reads}
- Most `insilico` data is used for testing software before using real data.
- Generating `insilico` data can be challenging but not can provide a starting data for testing some pipelines.
- Below is a demo for generating NovaSeq, MiSeq and HiSeq `insilico` sequencing data.

### Using NovaSeq platform
```{}
iss generate \
	--ncbi bacteria viruses archaea -U 15 0 2 \
	--coverage lognormal \
	--n_reads 0.12M \
	--model novaseq \annual report
	--output data/novaseq_01 \
	--compress \
	--seed 1
iss generate \
	--ncbi bacteria viruses archaea -U 15 1 1 \
	--coverage lognormal \
	--n_reads 0.11M \
	--model novaseq \
	--output data/novaseq_02 \
	--compress \
	--seed 2
iss generate \
	--ncbi bacteria viruses archaea -U 10 0 0 \
	--coverage lognormal \
	--n_reads 0.1M \
	--model novaseq \
	--output data/novaseq_03 \
	--compress \
	--seed 3
iss generate \
	--ncbi bacteria viruses archaea -U 10 0 0 \
	--coverage lognormal \
	--n_reads 0.1M \
	--model novaseq \
	--output data/novaseq_04 \
	--compress \
	--seed 4
iss generate \
	--ncbi bacteria viruses archaea -U 10 0 0 \
	--coverage lognormal \
	--n_reads 0.1M \
	--model novaseq \
	--output data/novaseq_05 \
	--compress \
	--seed 5
	
```	

### Using MiSeq platform
```{}
iss generate \
	--ncbi bacteria viruses archaea -U 10 0 2 \
	--coverage lognormal \
	--n_reads 0.12M \
	--model miseq \
	--output data/miseq_01 \
	--compress \
	--seed 16
iss generate \
	--ncbi bacteria viruses archaea -U 15 1 1 \
	--coverage lognormal \
	--n_reads 0.11M \
	--model miseq \
	--output data/miseq_02 \
	--compress \
	--seed 17
iss generate \
	--ncbi bacteria viruses archaea -U 5 0 0 \
	--coverage lognormal \
	--n_reads 0.1M \
	--model miseq \
	--output data/miseq_03 \
	--compress \
	--seed 18
iss generate \
	--ncbi bacteria viruses archaea -U 15 0 0 \
	--coverage lognormal \
	--n_reads 0.1M \
	--model miseq \
	--output data/miseq_04 \
	--compress \
	--seed 19
iss generate \
	--ncbi bacteria viruses archaea -U 5 0 0 \
	--coverage lognormal \
	--n_reads 0.1M \
	--model miseq \
	--output data/miseq_05 \
	--compress \
	--seed 20
	
```	
	
### Using HiSeq  platform

```{}
iss generate \
	--ncbi bacteria viruses archaea -U 10 1 2 \
	--coverage lognormal \
	--n_reads 0.12M \
	--model hiseq \
	--output data/hiseq_01 \
	--compress \
	--seed 21
iss generate \
	--ncbi bacteria viruses archaea -U 15 1 1 \
	--coverage lognormal \
	--n_reads 0.11M \
	--model hiseq \
	--output data/hiseq_02 \
	--compress \
	--seed 22
iss generate \
	--ncbi bacteria viruses archaea -U 10 0 0 \
	--coverage lognormal \
	--n_reads 0.1M \
	--model hiseq \
	--output data/hiseq_03 \
	--compress \
	--seed 23
iss generate \
	--ncbi bacteria viruses archaea -U 10 0 0 \
	--coverage lognormal \
	--n_reads 0.1M \
	--model hiseq \
	--output data/hiseq_04 \
	--compress \
	--seed 24
iss generate \
	--ncbi bacteria viruses archaea -U 10 0 0 \
	--coverage lognormal \
	--n_reads 0.1M \
	--model hiseq \
	--output data/hiseq_05 \
	--compress \
	--seed 25

```

<br>

## Resizing Fastq Files {#resize-fastq}
- Sometimes we want to extract a small subset to test the bioinformatics pipeline.
- You can resize the fastq files using the `seqkit sample` function [@Shen2016; @seqkit2022].
- Below is a demo for extracting only 0.1% of the sequencing data.

```bash
mkdir -p data
for i in {706..761}
  do
    cat SRR7450$i\_R1.fastq \
    | seqkit sample -p 0.01 \
    | seqkit shuffle -o data/SRR7450$i\_R1_sub.fastq \
    | cat SRR7450$i\_R2.fastq \
    | seqkit sample -p 0.01 \
    | seqkit shuffle -o data/SRR7450$i\_R2_sub.fastq
  done

```


