---
title: "End-to-End Microbiome Data Analysis in R"
subtitle: "PART 1: Preparing for Microbiome Data Analysis"
author: "Teresia Mrema-Buza, A Microbiome Computational Scientist and Owner of the Complex Data Insights, LLC, USA"
date: "`r Sys.Date()`"
site: bookdown::bookdown_site
knit: bookdown::render_book
output:
  bookdown::bs4_book:
    includes:
      in_header:
        
  bookdown::gitbook:
    includes:
      in_header: header.html
    config: 
      toc:
       collapse: 
      search: yes
      fontsettings:
        size: 2
    split_by: section
    split_bib: no
    df_print: paged
    number_sections: true
    css:
    - css/style.css
    - style.css
    
  html_document:
    code_folding: hide
    
  bookdown::html_document2:
    code_folding: hide
    
  bookdown::pdf_book:
    config:
      toc = true
      toc_depth = 2
      number_sections = true
      fig_caption = true
      keep_tex = no
      pandoc_args = null
      toc_unnumbered = no
      toc_appendix = no
      toc_bib = no 
      quote_footer = null
      highlight_bw = true
      latex_engine = xelatex
      df_print = kable
      base_format = rmarkdown::pdf_document
  includes:
    in_header: 
    - latex/header.tex
    - latex/preamble.tex
documentclass: book
classoption: openany #remove empty pages in pdf doc
bibliography:
- library/book.bib
- library/packages.bib
- library/microbiome.bib
- library/software.bib 
citation_package:
- natbib
- biblatex
- amsplain
colorlinks: true
css:
- css/style.css
always_allow_html: true
fig_caption: true
fontsize: 12pt
geometry: margin=1in
indent: false
keep_tex: true
link-citations: true
mainfont: Times New Roman
biblio-style: apalike
spacing: double
header-includes: 
- \usepackage{setspace}
- \newpage
- \newenvironment{tmbinfo}[0]{}{}
- \renewenvironment{tmbinfo}[0]{}{}
- \newenvironment{tmbalert}[0]{}{}
- \renewenvironment{tmbalert}[0]{}{}
- \newenvironment{tmbshare}[0]{}{}
- \renewenvironment{tmbshare}[0]{}{}
description: |
  This is a practical user's guide for **Systematic Microbiome Data Analysis in R**. The guide provides integrated and highly curated solutions for achieving better results.
---


```{r pkgbiblib, include=FALSE}
knitr::write_bib(c(
  .packages(), 'base','bookdown','rmarkdown','tidyverse','shiny','vegan','data.table, dendextend, robCompositions, microbiome, ALDEx2, caret, rms, phyloseq'
), 'library/packages.bib')
```

```{r setup, include=FALSE}
source(file = "R/common.R")
```

# Getting Started with Microbiome Data Analysis in R {-#frontpage}

```{r include=FALSE}
library(DiagrammeR)
library(DiagrammeRsvg)
library(rsvg)

# 1. Make a graph
graph <- DiagrammeR::grViz("
digraph {
graph [layout = dot, rankdir = TD]

node [
shape = box, 
style = filled, 
fillcolor = white, 
fontname = Helvetica,
penwidth = 2.0] 

edge [arrowhead = diamond]

A [label = 'Getting Started', fillcolor = white, penwidth = 5.0]
B [label = 'Requirements']
C [label = 'Basic Software']
D [label = 'Sample Metadata']
E [label = 'Sequencing Data']
F [label = 'Analysis & Visualization Tools']
G [label = 'Preprocessing Tools']
H [label = 'Bioinformatics Pipelines']
I [label = 'Mapping Files']

{A}  -> B
{B}  -> C
{B}  -> D
{B}  -> E
{C}  -> F
{C}  -> G
{C}  -> H
{B}  -> I

}", height = 400, width = 600)

# 2. Convert to SVG, then save as png
part1 = DiagrammeRsvg::export_svg(graph)
part1 = charToRaw(part1) # flatten
rsvg::rsvg_png(part1, "img/part1.png")
```


<a href=""><img src="images/part1_6-tiers.png" alt="Book cover" width="100%" style="padding: 50px 0px 100px 0px; float: right;"/></a>

## Quick Glimpse {-}
Investigating the role of microbial communities in health and disease requires a thorough knowledge of the entire analytical process. Using wrong approaches can cost a significant amount of dollars and lengthy process to achieve the desired results. This is <b>PART 1</b> of the practical user guides intended to provide analytical support to the microbiome research community. The entire guide is reproducible, allowing users to easily follow along. If interested, user may use this model to publish their findings in a book format.

## Structure of this guide {-}
This guide is divided into chapters to facilitate easy navigation. Each chapter contains several sections as displayed in the navigation bars on the left and right. Click the hyper-linked text if you want to jump into a specific chapter or section.

## Code availability {-}
The code is available at a public [GitHub repository](https://github.com/tmbuza/microbiome-part1/). If interested you can request a consulting service by contacting the developer of this repo using <a href="https://complexdatainsights.com/contact-us">this contact form</a>. 