<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chapter 5 Bioinformatics Pipelines | Getting Started with Microbiome Data Analysis in R</title>
<meta name="author" content="Teresia Mrema-Buza, A Microbiome Computational Scientist and Owner of the Complex Data Insights, LLC, USA">
<meta name="description" content="5.1 Pipeline for analyzing 16S rRNA amplicon data Below are few most popular platforms that offers free lightweight and reproducible tutorials for analyzing 16S rRNA gene sequencing data. Pipeline...">
<meta name="generator" content="bookdown 0.25 with bs4_book()">
<meta property="og:title" content="Chapter 5 Bioinformatics Pipelines | Getting Started with Microbiome Data Analysis in R">
<meta property="og:type" content="book">
<meta property="og:description" content="5.1 Pipeline for analyzing 16S rRNA amplicon data Below are few most popular platforms that offers free lightweight and reproducible tutorials for analyzing 16S rRNA gene sequencing data. Pipeline...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Chapter 5 Bioinformatics Pipelines | Getting Started with Microbiome Data Analysis in R">
<meta name="twitter:description" content="5.1 Pipeline for analyzing 16S rRNA amplicon data Below are few most popular platforms that offers free lightweight and reproducible tutorials for analyzing 16S rRNA gene sequencing data. Pipeline...">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><link href="libs/_Sen-0.4.1/font.css" rel="stylesheet">
<link href="https://fonts.googleapis.com/css2?family=DM%20Mono&amp;display=swap" rel="stylesheet">
<link href="libs/_Bitter-0.4.1/font.css" rel="stylesheet">
<script src="libs/bs3compat-0.3.1/transition.js"></script><script src="libs/bs3compat-0.3.1/tabs.js"></script><script src="libs/bs3compat-0.3.1/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="libs/htmlwidgets-1.5.4/htmlwidgets.js"></script><script src="libs/viz-1.8.2/viz.js"></script><link href="libs/DiagrammeR-styles-0.2/styles.css" rel="stylesheet">
<script src="libs/grViz-binding-1.0.9/grViz.js"></script><link href="libs/leaflet-1.3.1/leaflet.css" rel="stylesheet">
<script src="libs/leaflet-1.3.1/leaflet.js"></script><link href="libs/leafletfix-1.0.0/leafletfix.css" rel="stylesheet">
<script src="libs/proj4-2.6.2/proj4.min.js"></script><script src="libs/Proj4Leaflet-1.0.1/proj4leaflet.js"></script><link href="libs/rstudio_leaflet-1.3.1/rstudio_leaflet.css" rel="stylesheet">
<script src="libs/leaflet-binding-2.1.1/leaflet.js"></script><script src="libs/leaflet-providers-1.9.0/leaflet-providers_1.9.0.js"></script><script src="libs/leaflet-providers-plugin-2.1.1/leaflet-providers-plugin.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<style type="text/css">
    /* Used with Pandoc 2.11+ new --citeproc when CSL is used */
    div.csl-bib-body { }
    div.csl-entry {
      clear: both;
        }
    .hanging div.csl-entry {
      margin-left:2em;
      text-indent:-2em;
    }
    div.csl-left-margin {
      min-width:2em;
      float:left;
    }
    div.csl-right-inline {
      margin-left:2em;
      padding-left:1em;
    }
    div.csl-indent {
      margin-left: 2em;
    }
  </style>
<link rel="stylesheet" href="css/style.css">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="Integrated solutions for better results">Getting Started with Microbiome Data Analysis in R</a>:
        <small class="text-muted">Integrated solutions for better results</small>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html">Getting Started with Microbiome Data Analysis in R</a></li>
<li class="book-part">GETTING STARTED</li>
<li><a class="" href="basic-software.html"><span class="header-section-number">1</span> Basic Software</a></li>
<li><a class="" href="sample-metadata.html"><span class="header-section-number">2</span> Sample Metadata</a></li>
<li><a class="" href="sequencing-data.html"><span class="header-section-number">3</span> Sequencing Data</a></li>
<li><a class="" href="preprocessing-tools.html"><span class="header-section-number">4</span> Read Processing Tools</a></li>
<li><a class="active" href="bioinformatics-pipelines.html"><span class="header-section-number">5</span> Bioinformatics Pipelines</a></li>
<li class="book-part">APPENDIX</li>
<li><a class="" href="saved-data-objects.html"><span class="header-section-number">A</span> Saved Data Objects</a></li>
<li><a class="" href="software-and-packages-used.html"><span class="header-section-number">B</span> Software and Packages Used</a></li>
<li><a class="" href="references.html">References</a></li>
</ul>

        <div class="book-extra">
          <p><a id="book-repo" href="https://github.com/tmbuza/microbiome-part1">View book source <i class="fas fa-air-freshener"></i></a></p>
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="bioinformatics-pipelines" class="section level1" number="5">
<h1>
<span class="header-section-number">5</span> Bioinformatics Pipelines<a class="anchor" aria-label="anchor" href="#bioinformatics-pipelines"><i class="fas fa-link"></i></a>
</h1>
<div id="amplicon-analysis" class="section level2" number="5.1">
<h2>
<span class="header-section-number">5.1</span> Pipeline for analyzing 16S rRNA amplicon data<a class="anchor" aria-label="anchor" href="#amplicon-analysis"><i class="fas fa-link"></i></a>
</h2>
<p>Below are few most popular platforms that offers free lightweight and reproducible tutorials for analyzing 16S rRNA gene sequencing data.</p>
<div class="inline-table"><table class="table table-sm">
<colgroup>
<col width="22%">
<col width="34%">
<col width="44%">
</colgroup>
<thead><tr class="header">
<th>Pipeline</th>
<th>Description</th>
<th>Tutorial</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>Mothur</td>
<td>An open-source software package for bioinformatics data processing including 16S rRNA gene sequencing data</td>
<td><a href="https://mothur.org/wiki/miseq_sop/">Basic Tutorial</a></td>
</tr>
<tr class="even">
<td>QIIME2</td>
<td>An open-source microbiome analysis package including 16S rRNA gene sequencing data</td>
<td><a href="https://docs.qiime2.org/2022.2/tutorials/">Tutorials</a></td>
</tr>
<tr class="odd">
<td>Nephele</td>
<td>Open, accessible high performmance cloud platform platform for microbial bioinformatics. Integrates Mothur, QIIME2, DADA2 and Biobakery</td>
<td><a href="https://nephele.niaid.nih.gov/user_guide/">Some tutorials</a></td>
</tr>
<tr class="even">
<td></td>
<td></td>
<td></td>
</tr>
</tbody>
</table></div>
<blockquote>
<p>Make sure to check for latest instructions online.</p>
</blockquote>
</div>
<div id="metagenomics-analysis" class="section level2" number="5.2">
<h2>
<span class="header-section-number">5.2</span> Pipelines for analyzing metagenomics data<a class="anchor" aria-label="anchor" href="#metagenomics-analysis"><i class="fas fa-link"></i></a>
</h2>
<p>The installation instruction of various pipeline is found <a href="https://github.com/biobakery/biobakery/wiki">here</a>. See the table below for specific software.</p>
<div class="inline-table"><table class="table table-sm">
<colgroup>
<col width="22%">
<col width="34%">
<col width="44%">
</colgroup>
<thead><tr class="header">
<th>Pipeline</th>
<th>Description</th>
<th>Tutorial</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>Keaddata</td>
<td>Quality controll on metagenomic sequencing data</td>
<td><a href="https://github.com/biobakery/biobakery/wiki/kneaddata">Tutorial</a></td>
</tr>
<tr class="even">
<td>MetaPhlAn</td>
<td>Taxonomic Profiling from metagenomic shotgun sequencing data</td>
<td><a href="https://github.com/biobakery/biobakery/wiki/metaphlan3">Tutorial</a></td>
</tr>
<tr class="odd">
<td>HumanN</td>
<td>Functional Profiling from metagenomic sequencing data</td>
<td><a href="https://github.com/biobakery/biobakery/wiki/humann3">Tutorial</a></td>
</tr>
<tr class="even">
<td></td>
<td></td>
<td></td>
</tr>
</tbody>
</table></div>
<blockquote>
<p>Make sure to check for latest instructions online.</p>
</blockquote>

</div>
</div>



  <div class="chapter-nav">
<div class="prev"><a href="preprocessing-tools.html"><span class="header-section-number">4</span> Read Processing Tools</a></div>
<div class="next"><a href="saved-data-objects.html"><span class="header-section-number">A</span> Saved Data Objects</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#bioinformatics-pipelines"><span class="header-section-number">5</span> Bioinformatics Pipelines</a></li>
<li><a class="nav-link" href="#amplicon-analysis"><span class="header-section-number">5.1</span> Pipeline for analyzing 16S rRNA amplicon data</a></li>
<li><a class="nav-link" href="#metagenomics-analysis"><span class="header-section-number">5.2</span> Pipelines for analyzing metagenomics data</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
<li><a id="book-source" href="https://github.com/tmbuza/microbiome-part1/blob/main/01e_bioinfo_pipelines.Rmd">View source <i class="fas fa-air-freshener"></i></a></li>
          <li><a id="book-edit" href="https://github.com/tmbuza/microbiome-part1/edit/main/01e_bioinfo_pipelines.Rmd">Edit this page <i class="fas fa-air-freshener"></i></a></li>
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>Getting Started with Microbiome Data Analysis in R</strong>: Integrated solutions for better results" was written by Teresia Mrema-Buza, A Microbiome Computational Scientist and Owner of the Complex Data Insights, LLC, USA. It was last built on 2022-05-17.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer>
</body>
</html>
