<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chapter 4 Read Processing Tools | Getting Started with Microbiome Data Analysis in R</title>
<meta name="author" content="Teresia Mrema-Buza, A Microbiome Computational Scientist and Owner of the Complex Data Insights, LLC, USA">
<meta name="description" content="There several tools out there that can help in preprocessing raw read. Listed below are some of most common tools used in understanding the characteristics of the read and their quality scores....">
<meta name="generator" content="bookdown 0.25 with bs4_book()">
<meta property="og:title" content="Chapter 4 Read Processing Tools | Getting Started with Microbiome Data Analysis in R">
<meta property="og:type" content="book">
<meta property="og:description" content="There several tools out there that can help in preprocessing raw read. Listed below are some of most common tools used in understanding the characteristics of the read and their quality scores....">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Chapter 4 Read Processing Tools | Getting Started with Microbiome Data Analysis in R">
<meta name="twitter:description" content="There several tools out there that can help in preprocessing raw read. Listed below are some of most common tools used in understanding the characteristics of the read and their quality scores....">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><link href="libs/_Sen-0.4.1/font.css" rel="stylesheet">
<link href="https://fonts.googleapis.com/css2?family=DM%20Mono&amp;display=swap" rel="stylesheet">
<link href="libs/_Bitter-0.4.1/font.css" rel="stylesheet">
<script src="libs/bs3compat-0.3.1/transition.js"></script><script src="libs/bs3compat-0.3.1/tabs.js"></script><script src="libs/bs3compat-0.3.1/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="libs/htmlwidgets-1.5.4/htmlwidgets.js"></script><script src="libs/viz-1.8.2/viz.js"></script><link href="libs/DiagrammeR-styles-0.2/styles.css" rel="stylesheet">
<script src="libs/grViz-binding-1.0.9/grViz.js"></script><link href="libs/leaflet-1.3.1/leaflet.css" rel="stylesheet">
<script src="libs/leaflet-1.3.1/leaflet.js"></script><link href="libs/leafletfix-1.0.0/leafletfix.css" rel="stylesheet">
<script src="libs/proj4-2.6.2/proj4.min.js"></script><script src="libs/Proj4Leaflet-1.0.1/proj4leaflet.js"></script><link href="libs/rstudio_leaflet-1.3.1/rstudio_leaflet.css" rel="stylesheet">
<script src="libs/leaflet-binding-2.1.1/leaflet.js"></script><script src="libs/leaflet-providers-1.9.0/leaflet-providers_1.9.0.js"></script><script src="libs/leaflet-providers-plugin-2.1.1/leaflet-providers-plugin.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<style type="text/css">
    /* Used with Pandoc 2.11+ new --citeproc when CSL is used */
    div.csl-bib-body { }
    div.csl-entry {
      clear: both;
        }
    .hanging div.csl-entry {
      margin-left:2em;
      text-indent:-2em;
    }
    div.csl-left-margin {
      min-width:2em;
      float:left;
    }
    div.csl-right-inline {
      margin-left:2em;
      padding-left:1em;
    }
    div.csl-indent {
      margin-left: 2em;
    }
  </style>
<link rel="stylesheet" href="css/style.css">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="Integrated solutions for better results">Getting Started with Microbiome Data Analysis in R</a>:
        <small class="text-muted">Integrated solutions for better results</small>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html">Getting Started with Microbiome Data Analysis in R</a></li>
<li class="book-part">GETTING STARTED</li>
<li><a class="" href="basic-software.html"><span class="header-section-number">1</span> Basic Software</a></li>
<li><a class="" href="sample-metadata.html"><span class="header-section-number">2</span> Sample Metadata</a></li>
<li><a class="" href="sequencing-data.html"><span class="header-section-number">3</span> Sequencing Data</a></li>
<li><a class="active" href="preprocessing-tools.html"><span class="header-section-number">4</span> Read Processing Tools</a></li>
<li><a class="" href="bioinformatics-pipelines.html"><span class="header-section-number">5</span> Bioinformatics Pipelines</a></li>
<li class="book-part">APPENDIX</li>
<li><a class="" href="saved-data-objects.html"><span class="header-section-number">A</span> Saved Data Objects</a></li>
<li><a class="" href="software-and-packages-used.html"><span class="header-section-number">B</span> Software and Packages Used</a></li>
<li><a class="" href="references.html">References</a></li>
</ul>

        <div class="book-extra">
          <p><a id="book-repo" href="https://github.com/tmbuza/microbiome-part1">View book source <i class="fas fa-air-freshener"></i></a></p>
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="preprocessing-tools" class="section level1" number="4">
<h1>
<span class="header-section-number">4</span> Read Processing Tools<a class="anchor" aria-label="anchor" href="#preprocessing-tools"><i class="fas fa-link"></i></a>
</h1>
<p>There several tools out there that can help in preprocessing raw read. Listed below are some of most common tools used in understanding the characteristics of the read and their quality scores. Click on the hyperlinked tool to learn more how to install it.</p>
<ul>
<li>
<a href="https://bioinf.shenwei.me/seqkit/download/">Seqkit</a> <span class="citation">(<a href="references.html#ref-seqkit2016" role="doc-biblioref">Wei Shen et al. 2016</a>)</span>
</li>
<li>
<a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a> <span class="citation">(<a href="references.html#ref-fastqc2018" role="doc-biblioref">Andrews 2018</a>)</span>
</li>
<li>
<a href="https://multiqc.info/">MultiQC</a> <span class="citation">(<a href="references.html#ref-multiqc2016" role="doc-biblioref">Ewels et al. 2016</a>)</span>
</li>
<li>
<a href="https://sourceforge.net/projects/bbmap/files/latest/download">BBMap</a> platform <span class="citation">(<a href="references.html#ref-bbmap2016" role="doc-biblioref">Bushnell 2016</a>)</span>
</li>
<li>
<a href="https://huttenhower.sph.harvard.edu/kneaddata/">Kneaddata</a> in biobakery platform <span class="citation">(<a href="references.html#ref-kneaddata2022" role="doc-biblioref">Biobakery 2022</a>)</span>
</li>
</ul>
<div id="example-installing-on-mac" class="section level2" number="4.1">
<h2>
<span class="header-section-number">4.1</span> Example installing on Mac<a class="anchor" aria-label="anchor" href="#example-installing-on-mac"><i class="fas fa-link"></i></a>
</h2>
<div id="install-seqkit" class="section level3" number="4.1.1">
<h3>
<span class="header-section-number">4.1.1</span> Install seqkit<a class="anchor" aria-label="anchor" href="#install-seqkit"><i class="fas fa-link"></i></a>
</h3>
<div class="sourceCode" id="cb35"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb35-1"><a href="preprocessing-tools.html#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="fu">wget</span> <span class="at">--no-check-certificate</span> https://github.com/shenwei356/seqkit/releases/download/v0.8.0/seqkit_darwin_amd64.tar.gz</span>
<span id="cb35-2"><a href="preprocessing-tools.html#cb35-2" aria-hidden="true" tabindex="-1"></a><span class="fu">tar</span> <span class="at">-zxvf</span> seqkit_darwin_amd64.tar.gz</span>
<span id="cb35-3"><a href="preprocessing-tools.html#cb35-3" aria-hidden="true" tabindex="-1"></a><span class="fu">mv</span> seqkit code/</span>
<span id="cb35-4"><a href="preprocessing-tools.html#cb35-4" aria-hidden="true" tabindex="-1"></a><span class="fu">cp</span> code/seqkit ~/bin/</span>
<span id="cb35-5"><a href="preprocessing-tools.html#cb35-5" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span> seqkit_darwin_amd64.tar.gz</span></code></pre></div>
</div>
<div id="install-fastqc" class="section level3" number="4.1.2">
<h3>
<span class="header-section-number">4.1.2</span> Install fastqc<a class="anchor" aria-label="anchor" href="#install-fastqc"><i class="fas fa-link"></i></a>
</h3>
<div class="sourceCode" id="cb36"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb36-1"><a href="preprocessing-tools.html#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="fu">wget</span> <span class="at">--no-check-certificate</span> https://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.11.7.zip</span>
<span id="cb36-2"><a href="preprocessing-tools.html#cb36-2" aria-hidden="true" tabindex="-1"></a><span class="fu">unzip</span> fastqc_v0.11.7.zip</span>
<span id="cb36-3"><a href="preprocessing-tools.html#cb36-3" aria-hidden="true" tabindex="-1"></a><span class="fu">chmod</span> 755 FastQC/fastqc</span>
<span id="cb36-4"><a href="preprocessing-tools.html#cb36-4" aria-hidden="true" tabindex="-1"></a><span class="fu">mv</span> <span class="at">-v</span> FastQC code/</span>
<span id="cb36-5"><a href="preprocessing-tools.html#cb36-5" aria-hidden="true" tabindex="-1"></a><span class="fu">cp</span> code/FastQC/fastqc ~/bin/</span>
<span id="cb36-6"><a href="preprocessing-tools.html#cb36-6" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span> fastqc_v0.11.7.zip</span></code></pre></div>
</div>
<div id="install-bbmap" class="section level3" number="4.1.3">
<h3>
<span class="header-section-number">4.1.3</span> Install bbmap<a class="anchor" aria-label="anchor" href="#install-bbmap"><i class="fas fa-link"></i></a>
</h3>
<div class="sourceCode" id="cb37"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb37-1"><a href="preprocessing-tools.html#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="fu">wget</span> <span class="at">--no-check-certificate</span> https://sourceforge.net/projects/bbmap/files/BBMap_37.90.tar.gz</span>
<span id="cb37-2"><a href="preprocessing-tools.html#cb37-2" aria-hidden="true" tabindex="-1"></a><span class="fu">tar</span> <span class="at">-xvzf</span> BBMap_37.90.tar.gz</span>
<span id="cb37-3"><a href="preprocessing-tools.html#cb37-3" aria-hidden="true" tabindex="-1"></a><span class="fu">mv</span> <span class="at">-v</span> bbmap code/</span>
<span id="cb37-4"><a href="preprocessing-tools.html#cb37-4" aria-hidden="true" tabindex="-1"></a><span class="fu">cp</span> code/bbmap/bbduk.sh ~/bin/</span>
<span id="cb37-5"><a href="preprocessing-tools.html#cb37-5" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span> BBMap_37.90.tar.gz</span></code></pre></div>
<blockquote>
<p>Make sure to check for latest instructions online.</p>
</blockquote>

</div>
</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="sequencing-data.html"><span class="header-section-number">3</span> Sequencing Data</a></div>
<div class="next"><a href="bioinformatics-pipelines.html"><span class="header-section-number">5</span> Bioinformatics Pipelines</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#preprocessing-tools"><span class="header-section-number">4</span> Read Processing Tools</a></li>
<li>
<a class="nav-link" href="#example-installing-on-mac"><span class="header-section-number">4.1</span> Example installing on Mac</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#install-seqkit"><span class="header-section-number">4.1.1</span> Install seqkit</a></li>
<li><a class="nav-link" href="#install-fastqc"><span class="header-section-number">4.1.2</span> Install fastqc</a></li>
<li><a class="nav-link" href="#install-bbmap"><span class="header-section-number">4.1.3</span> Install bbmap</a></li>
</ul>
</li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
<li><a id="book-source" href="https://github.com/tmbuza/microbiome-part1/blob/main/01d_read_preprocessing_tools.Rmd">View source <i class="fas fa-air-freshener"></i></a></li>
          <li><a id="book-edit" href="https://github.com/tmbuza/microbiome-part1/edit/main/01d_read_preprocessing_tools.Rmd">Edit this page <i class="fas fa-air-freshener"></i></a></li>
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>Getting Started with Microbiome Data Analysis in R</strong>: Integrated solutions for better results" was written by Teresia Mrema-Buza, A Microbiome Computational Scientist and Owner of the Complex Data Insights, LLC, USA. It was last built on 2022-05-17.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer>
</body>
</html>
