<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chapter 1 Basic Software | Getting Started with Microbiome Data Analysis in R</title>
<meta name="author" content="Teresia Mrema-Buza, A Microbiome Computational Scientist and Owner of the Complex Data Insights, LLC, USA">
<meta name="description" content="1.1 Install R Environments  R is a free software for statistical computing, data analysis, and graphics (R Core Team 2021). We need to install R application on a personal computer to process the R...">
<meta name="generator" content="bookdown 0.25 with bs4_book()">
<meta property="og:title" content="Chapter 1 Basic Software | Getting Started with Microbiome Data Analysis in R">
<meta property="og:type" content="book">
<meta property="og:description" content="1.1 Install R Environments  R is a free software for statistical computing, data analysis, and graphics (R Core Team 2021). We need to install R application on a personal computer to process the R...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Chapter 1 Basic Software | Getting Started with Microbiome Data Analysis in R">
<meta name="twitter:description" content="1.1 Install R Environments  R is a free software for statistical computing, data analysis, and graphics (R Core Team 2021). We need to install R application on a personal computer to process the R...">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><link href="libs/_Sen-0.4.1/font.css" rel="stylesheet">
<link href="https://fonts.googleapis.com/css2?family=DM%20Mono&amp;display=swap" rel="stylesheet">
<link href="libs/_Bitter-0.4.1/font.css" rel="stylesheet">
<script src="libs/bs3compat-0.3.1/transition.js"></script><script src="libs/bs3compat-0.3.1/tabs.js"></script><script src="libs/bs3compat-0.3.1/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="libs/htmlwidgets-1.5.4/htmlwidgets.js"></script><script src="libs/viz-1.8.2/viz.js"></script><link href="libs/DiagrammeR-styles-0.2/styles.css" rel="stylesheet">
<script src="libs/grViz-binding-1.0.9/grViz.js"></script><link href="libs/leaflet-1.3.1/leaflet.css" rel="stylesheet">
<script src="libs/leaflet-1.3.1/leaflet.js"></script><link href="libs/leafletfix-1.0.0/leafletfix.css" rel="stylesheet">
<script src="libs/proj4-2.6.2/proj4.min.js"></script><script src="libs/Proj4Leaflet-1.0.1/proj4leaflet.js"></script><link href="libs/rstudio_leaflet-1.3.1/rstudio_leaflet.css" rel="stylesheet">
<script src="libs/leaflet-binding-2.1.1/leaflet.js"></script><script src="libs/leaflet-providers-1.9.0/leaflet-providers_1.9.0.js"></script><script src="libs/leaflet-providers-plugin-2.1.1/leaflet-providers-plugin.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<style type="text/css">
    /* Used with Pandoc 2.11+ new --citeproc when CSL is used */
    div.csl-bib-body { }
    div.csl-entry {
      clear: both;
        }
    .hanging div.csl-entry {
      margin-left:2em;
      text-indent:-2em;
    }
    div.csl-left-margin {
      min-width:2em;
      float:left;
    }
    div.csl-right-inline {
      margin-left:2em;
      padding-left:1em;
    }
    div.csl-indent {
      margin-left: 2em;
    }
  </style>
<link rel="stylesheet" href="css/style.css">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="Integrated solutions for better results">Getting Started with Microbiome Data Analysis in R</a>:
        <small class="text-muted">Integrated solutions for better results</small>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html">Getting Started with Microbiome Data Analysis in R</a></li>
<li class="book-part">GETTING STARTED</li>
<li><a class="active" href="basic-software.html"><span class="header-section-number">1</span> Basic Software</a></li>
<li><a class="" href="sample-metadata.html"><span class="header-section-number">2</span> Sample Metadata</a></li>
<li><a class="" href="sequencing-data.html"><span class="header-section-number">3</span> Sequencing Data</a></li>
<li><a class="" href="preprocessing-tools.html"><span class="header-section-number">4</span> Read Processing Tools</a></li>
<li><a class="" href="bioinformatics-pipelines.html"><span class="header-section-number">5</span> Bioinformatics Pipelines</a></li>
<li class="book-part">APPENDIX</li>
<li><a class="" href="saved-data-objects.html"><span class="header-section-number">A</span> Saved Data Objects</a></li>
<li><a class="" href="software-and-packages-used.html"><span class="header-section-number">B</span> Software and Packages Used</a></li>
<li><a class="" href="references.html">References</a></li>
</ul>

        <div class="book-extra">
          <p><a id="book-repo" href="https://github.com/tmbuza/microbiome-part1">View book source <i class="fas fa-air-freshener"></i></a></p>
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="basic-software" class="section level1" number="1">
<h1>
<span class="header-section-number">1</span> Basic Software<a class="anchor" aria-label="anchor" href="#basic-software"><i class="fas fa-link"></i></a>
</h1>
<div id="Renv" class="section level2" number="1.1">
<h2>
<span class="header-section-number">1.1</span> Install R Environments<a class="anchor" aria-label="anchor" href="#Renv"><i class="fas fa-link"></i></a>
</h2>
<p><a href=""><img src="images/rlogoblue.png" alt="R Software" width="100" style="padding: 0 15px; float: left;"></a>
<strong>R</strong> is a free software for statistical computing, data analysis, and graphics <span class="citation">(<a href="references.html#ref-RCoreTeam2021" role="doc-biblioref">R Core Team 2021</a>)</span>. We need to install R application on a personal computer to process the R programming language. You can download and install R using these steps:</p>
<ol style="list-style-type: decimal">
<li>Go to <a href="https://www.r-project.org/">https://www.r-project.org/</a>.</li>
<li>On the left, under Download, click on <a href="http://cran.r-project.org/mirrors.html">CRAN</a> to access the mirrors. CRAN (Comprehensive R Archive Network) is mirrored on nearly 100 registered servers in nearly 50 regions world. See <a href="https://cran.r-project.org/mirmon_report.html">CRAN mirror status</a>.</li>
<li>
<a href="https://cloud.r-project.org/" class="uri">https://cloud.r-project.org/</a> Pick a mirror that is close to your location, and automatically R will connect to that server ready to download the package files.</li>
<li>Select a compatible platform to download precompiled binary distributions of the base system, which also come with contributed packages.</li>
</ol>
<p><br></p>
</div>
<div id="rstudio" class="section level2" number="1.2">
<h2>
<span class="header-section-number">1.2</span> Install RStudio Environments<a class="anchor" aria-label="anchor" href="#rstudio"><i class="fas fa-link"></i></a>
</h2>
<p><a href=""><img src="images/rstudio.png" alt="RStudio" width="100" style="padding: 0 15px; float: left;"></a></p>
<p><strong>RStudio</strong> is a free program that integrates with R as an IDE (Integrated Development Environment) to implement most of the analytical functionalities <span class="citation">(<a href="references.html#ref-RStudioTeam2021" role="doc-biblioref">RStudio Team 2021</a>)</span>. For effective analysis, we must install R before installing RStudio. We will intensively use RStudio IDE to give us a user interface. We are interested in <strong>RStudio Desktop</strong>, which is the open-source regular desktop application. You can install it like this:</p>
<ol style="list-style-type: decimal">
<li>Go to <a href="https://rstudio.com/products/rstudio/">https://rstudio.com/products/rstudio/</a>.</li>
<li>Click on <a href="https://rstudio.com/products/rstudio/#rstudio-desktop">RStudio Desktop</a> box to move to the open source edition.</li>
<li>Choose your preferred <a href="https://rstudio.com/products/rstudio/download/">license</a> either open source or commercial.</li>
<li>Select <a href="https://rstudio.com/products/rstudio/download/#download">installer</a> compatible to your operating system.</li>
</ol>
<p><br></p>
<div class="figure">
<img src="images/RStudioIDE.png" style="width:100.0%" alt=""><p class="caption">Screen shot of RStudio User Interface</p>
</div>
<p><br></p>
</div>
<div id="installpkgs" class="section level2" number="1.3">
<h2>
<span class="header-section-number">1.3</span> Install R packages<a class="anchor" aria-label="anchor" href="#installpkgs"><i class="fas fa-link"></i></a>
</h2>
<p>Install basic packages that get the analysis started. We will introduce additional packages within the subsequent sections whenever needed</p>
<p><strong>Description of selected packages</strong></p>
<div class="inline-table"><table class="table table-sm">
<colgroup>
<col width="31%">
<col width="68%">
</colgroup>
<thead><tr class="header">
<th align="left">Package</th>
<th align="left">Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left"><a href="https://www.rdocumentation.org/packages/tidyverse/">tidyverse</a></td>
<td align="left">The <em>tidyverse</em> package is a collection of R packages designed with common APIs. It includes <em>readr</em>, <em>dplyr</em>, <em>tidyr</em>, <em>ggplot2</em>, <em>tibble</em>, <em>purr</em>, etc.</td>
</tr>
<tr class="even">
<td align="left"><a href="https://www.rdocumentation.org/packages/readr/">readr</a></td>
<td align="left">The <em>readr</em> package turns flat files into data frames</td>
</tr>
<tr class="odd">
<td align="left"><a href="https://www.rdocumentation.org/packages/dplyr/">dplyr</a></td>
<td align="left">The <em>dplyr</em> package focuses on data frames. We will use dplyr (a lot) to subset, summarize, rearrange, and to join together data sets.</td>
</tr>
<tr class="even">
<td align="left"><a href="https://www.rdocumentation.org/packages/tidyr/">tidyr</a></td>
<td align="left">The <em>tidyr</em> package converts data into the tidy format where each variable is a column, each observation is a row, and each type of observational unit is a table.</td>
</tr>
<tr class="odd">
<td align="left"><a href="https://www.rdocumentation.org/packages/ggplot2/">ggplot2</a></td>
<td align="left">The <em>ggplot2</em> functions create elegant graphics based on the Grammar of Graphics.</td>
</tr>
<tr class="even">
<td align="left"><a href="https://www.rdocumentation.org/packages/knitr/">knitr</a></td>
<td align="left">The <em>knitr</em> package is excellect for integrating R code into different forms of Literate Programming to generate dynamic reports.</td>
</tr>
<tr class="odd">
<td align="left"><a href="https://www.rdocumentation.org/packages/magrittr/">magrittr</a></td>
<td align="left">The <em>magrittr</em> provides an operator ( %&gt;% ) for chaining several commands.</td>
</tr>
<tr class="even">
<td align="left"><a href="https://www.rdocumentation.org/packages/purrr/">purr</a></td>
<td align="left">The <em>purr</em> provides tools for working with functions and vectors.</td>
</tr>
<tr class="odd">
<td align="left"><a href="https://www.rdocumentation.org/packages/forcats/">forcat</a></td>
<td align="left">The <em>forcats</em> provide functions for solving common problems with factors.</td>
</tr>
</tbody>
</table></div>
<div class="tmbinfo">
<p>
<strong>Location of R packages used in this Book</strong>
</p>
<ul>
<li>
<a href="https://cran.r-project.org/">CRAN</a> (Comprehensive R Archive Network): The official repository. Reviewed!
</li>
<li>
<a href="https://www.bioconductor.org/">Bioconductor</a>: Most popular repository for bioinformatics software. Reviewed!
</li>
<li>
<a href="https://github.com/">Github</a>: Most popular repository for open source projects. Not Reviewed!
</li>
</ul>
</div>
<p><strong>Installing and loading packages from CRAN</strong></p>
<pre><code>install.packages("packagename")
library("packagename")</code></pre>
<p><strong>Installing and loading R packages from Bioconductor</strong></p>
<pre><code># Install BiocManager
if (!requireNamespace("BiocManager", quietly = TRUE)) install.packages("BiocManager")
library("BiocManager")

# Then install packages like this:
BiocManager::install("packagename")}
library("packagename")</code></pre>
<p><strong>Installing and loading R packages from GitHub</strong>.</p>
<pre><code># Install devtools from CRAN
install.packages("devtools")
library(devtools)

# Or the development version from GitHub:
devtools::install_github("r-lib/devtools")
library(devtools)

# Then install any GitHub package like this:

devtools::install_github("githubID/packagename")
library("packagename")</code></pre>
</div>
<div id="getting-help-in-r" class="section level2" number="1.4">
<h2>
<span class="header-section-number">1.4</span> Getting Help in R<a class="anchor" aria-label="anchor" href="#getting-help-in-r"><i class="fas fa-link"></i></a>
</h2>
<p><strong>Show basic package information</strong></p>
<pre><code>help("packagename")

or

?packagename

help(package = "packagename")
</code></pre>
<p><strong>Get a general overview of a function, including an example code</strong></p>
<pre><code>help(functionname)

or

help(functionname, package = "packagename")
</code></pre>
<p><strong>Remove unneeded packages from your computer</strong></p>
<pre><code>remove.packages("packagename")</code></pre>
<p><strong>Unload a package</strong></p>
<pre><code>detach("package::packagename")</code></pre>
</div>
<div id="citation-information" class="section level2" number="1.5">
<h2>
<span class="header-section-number">1.5</span> Citation Information<a class="anchor" aria-label="anchor" href="#citation-information"><i class="fas fa-link"></i></a>
</h2>
<p>Run on R console command line</p>
<pre><code><a href="https://rdrr.io/r/utils/citation.html"># Citing R\n
citation()

# Citing RStudio
library(rstudioapi)
versionInfo()$citation

# Citing Package ...example
citation("tidyverse")</a></code></pre>
</div>
<div id="useful-links" class="section level2" number="1.6">
<h2>
<span class="header-section-number">1.6</span> Useful links<a class="anchor" aria-label="anchor" href="#useful-links"><i class="fas fa-link"></i></a>
</h2>
<ol style="list-style-type: decimal">
<li>R Packages: <a href="https://www.datacamp.com/community/tutorials/r-packages-guide">https://www.datacamp.com/community/tutorials/r-packages-guide</a>.</li>
<li>Search CRAN, Bioconductor and GitHub packages and functions: <a href="https://www.rdocumentation.org/">https://www.rdocumentation.org/</a>.</li>
</ol>
</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="index.html">Getting Started with Microbiome Data Analysis in R</a></div>
<div class="next"><a href="sample-metadata.html"><span class="header-section-number">2</span> Sample Metadata</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#basic-software"><span class="header-section-number">1</span> Basic Software</a></li>
<li><a class="nav-link" href="#Renv"><span class="header-section-number">1.1</span> Install R Environments</a></li>
<li><a class="nav-link" href="#rstudio"><span class="header-section-number">1.2</span> Install RStudio Environments</a></li>
<li><a class="nav-link" href="#installpkgs"><span class="header-section-number">1.3</span> Install R packages</a></li>
<li><a class="nav-link" href="#getting-help-in-r"><span class="header-section-number">1.4</span> Getting Help in R</a></li>
<li><a class="nav-link" href="#citation-information"><span class="header-section-number">1.5</span> Citation Information</a></li>
<li><a class="nav-link" href="#useful-links"><span class="header-section-number">1.6</span> Useful links</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
<li><a id="book-source" href="https://github.com/tmbuza/microbiome-part1/blob/main/01a_basic_software.Rmd">View source <i class="fas fa-air-freshener"></i></a></li>
          <li><a id="book-edit" href="https://github.com/tmbuza/microbiome-part1/edit/main/01a_basic_software.Rmd">Edit this page <i class="fas fa-air-freshener"></i></a></li>
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>Getting Started with Microbiome Data Analysis in R</strong>: Integrated solutions for better results" was written by Teresia Mrema-Buza, A Microbiome Computational Scientist and Owner of the Complex Data Insights, LLC, USA. It was last built on 2022-05-17.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer>
</body>
</html>
