<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>B Software and Packages Used | Getting Started with Microbiome Data Analysis in R</title>
<meta name="author" content="Teresia Mrema-Buza, A Microbiome Computational Scientist and Owner of the Complex Data Insights, LLC, USA">
<meta name="description" content="B.1 Basic dependencies R version 4.1.3 (2022-03-10) tidyverse (v. 1.3.1) knitr (v. 1.38) rmarkdown (v. 2.13) bookdown (v. 0.25)  B.2 Available on machine used R version 4.1.3 (2022-03-10)...">
<meta name="generator" content="bookdown 0.25 with bs4_book()">
<meta property="og:title" content="B Software and Packages Used | Getting Started with Microbiome Data Analysis in R">
<meta property="og:type" content="book">
<meta property="og:description" content="B.1 Basic dependencies R version 4.1.3 (2022-03-10) tidyverse (v. 1.3.1) knitr (v. 1.38) rmarkdown (v. 2.13) bookdown (v. 0.25)  B.2 Available on machine used R version 4.1.3 (2022-03-10)...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="B Software and Packages Used | Getting Started with Microbiome Data Analysis in R">
<meta name="twitter:description" content="B.1 Basic dependencies R version 4.1.3 (2022-03-10) tidyverse (v. 1.3.1) knitr (v. 1.38) rmarkdown (v. 2.13) bookdown (v. 0.25)  B.2 Available on machine used R version 4.1.3 (2022-03-10)...">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><link href="libs/_Sen-0.4.1/font.css" rel="stylesheet">
<link href="https://fonts.googleapis.com/css2?family=DM%20Mono&amp;display=swap" rel="stylesheet">
<link href="libs/_Bitter-0.4.1/font.css" rel="stylesheet">
<script src="libs/bs3compat-0.3.1/transition.js"></script><script src="libs/bs3compat-0.3.1/tabs.js"></script><script src="libs/bs3compat-0.3.1/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="libs/htmlwidgets-1.5.4/htmlwidgets.js"></script><script src="libs/viz-1.8.2/viz.js"></script><link href="libs/DiagrammeR-styles-0.2/styles.css" rel="stylesheet">
<script src="libs/grViz-binding-1.0.9/grViz.js"></script><link href="libs/leaflet-1.3.1/leaflet.css" rel="stylesheet">
<script src="libs/leaflet-1.3.1/leaflet.js"></script><link href="libs/leafletfix-1.0.0/leafletfix.css" rel="stylesheet">
<script src="libs/proj4-2.6.2/proj4.min.js"></script><script src="libs/Proj4Leaflet-1.0.1/proj4leaflet.js"></script><link href="libs/rstudio_leaflet-1.3.1/rstudio_leaflet.css" rel="stylesheet">
<script src="libs/leaflet-binding-2.1.1/leaflet.js"></script><script src="libs/leaflet-providers-1.9.0/leaflet-providers_1.9.0.js"></script><script src="libs/leaflet-providers-plugin-2.1.1/leaflet-providers-plugin.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<style type="text/css">
    /* Used with Pandoc 2.11+ new --citeproc when CSL is used */
    div.csl-bib-body { }
    div.csl-entry {
      clear: both;
        }
    .hanging div.csl-entry {
      margin-left:2em;
      text-indent:-2em;
    }
    div.csl-left-margin {
      min-width:2em;
      float:left;
    }
    div.csl-right-inline {
      margin-left:2em;
      padding-left:1em;
    }
    div.csl-indent {
      margin-left: 2em;
    }
  </style>
<link rel="stylesheet" href="css/style.css">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="Integrated solutions for better results">Getting Started with Microbiome Data Analysis in R</a>:
        <small class="text-muted">Integrated solutions for better results</small>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html">Getting Started with Microbiome Data Analysis in R</a></li>
<li class="book-part">GETTING STARTED</li>
<li><a class="" href="basic-software.html"><span class="header-section-number">1</span> Basic Software</a></li>
<li><a class="" href="sample-metadata.html"><span class="header-section-number">2</span> Sample Metadata</a></li>
<li><a class="" href="sequencing-data.html"><span class="header-section-number">3</span> Sequencing Data</a></li>
<li><a class="" href="preprocessing-tools.html"><span class="header-section-number">4</span> Read Processing Tools</a></li>
<li><a class="" href="bioinformatics-pipelines.html"><span class="header-section-number">5</span> Bioinformatics Pipelines</a></li>
<li class="book-part">APPENDIX</li>
<li><a class="" href="saved-data-objects.html"><span class="header-section-number">A</span> Saved Data Objects</a></li>
<li><a class="active" href="software-and-packages-used.html"><span class="header-section-number">B</span> Software and Packages Used</a></li>
<li><a class="" href="references.html">References</a></li>
</ul>

        <div class="book-extra">
          <p><a id="book-repo" href="https://github.com/tmbuza/microbiome-part1">View book source <i class="fas fa-air-freshener"></i></a></p>
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="software-and-packages-used" class="section level1" number="7">
<h1>
<span class="header-section-number">B</span> Software and Packages Used<a class="anchor" aria-label="anchor" href="#software-and-packages-used"><i class="fas fa-link"></i></a>
</h1>
<div id="basic-dependencies" class="section level2" number="7.1">
<h2>
<span class="header-section-number">B.1</span> Basic dependencies<a class="anchor" aria-label="anchor" href="#basic-dependencies"><i class="fas fa-link"></i></a>
</h2>
<ul>
<li>R version 4.1.3 (2022-03-10)</li>
<li>
<code>tidyverse</code> (v. 1.3.1)</li>
<li>
<code>knitr</code> (v. 1.38)</li>
<li>
<code>rmarkdown</code> (v. 2.13)</li>
<li>
<code>bookdown</code> (v. 0.25)</li>
</ul>
<!-- * `ggpubr` (v. 0.4.0) --><!-- * `downlit` (v. 0.4.0) --><!-- * `phyloseq` (v. 1.38.0) --><!-- * `ape` (v. 5.6.2) --><!-- * `ggtext` (v. 0.1.1) --><!-- * `dendextend` (v. 1.15.2) --><!-- * `metagMisc` (v. 0.0.4) --><!-- * `cgwtools` (v. 3.3) -->
</div>
<div id="available-on-machine-used" class="section level2" number="7.2">
<h2>
<span class="header-section-number">B.2</span> Available on machine used<a class="anchor" aria-label="anchor" href="#available-on-machine-used"><i class="fas fa-link"></i></a>
</h2>
<pre><code>R version 4.1.3 (2022-03-10)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur/Monterey 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] data.table_1.14.2       leaflet.providers_1.9.0 leaflet.esri_1.0.0     
 [4] leaflet.extras_1.0.0    leaflet_2.1.1           funModeling_1.9.4      
 [7] Hmisc_4.6-0             Formula_1.2-4           survival_3.3-1         
[10] lattice_0.20-45         scales_1.1.1            magrittr_2.0.3         
[13] DiagrammeRsvg_0.1       DiagrammeR_1.0.9        bookdown_0.25          
[16] rmarkdown_2.13          knitr_1.38              forcats_0.5.1          
[19] stringr_1.4.0           dplyr_1.0.8             purrr_0.3.4            
[22] readr_2.1.2             tidyr_1.2.0             tibble_3.1.6           
[25] ggplot2_3.3.5           tidyverse_1.3.1        

loaded via a namespace (and not attached):
 [1] fs_1.5.2            lubridate_1.8.0     bit64_4.0.5        
 [4] RColorBrewer_1.1-2  httr_1.4.2          tools_4.1.3        
 [7] backports_1.4.1     bslib_0.3.1         utf8_1.2.2         
[10] R6_2.5.1            rpart_4.1.16        lazyeval_0.2.2     
[13] DBI_1.1.2           colorspace_2.0-3    nnet_7.3-17        
[16] withr_2.5.0         gridExtra_2.3       tidyselect_1.1.2   
[19] moments_0.14        downlit_0.4.0       bit_4.0.4          
[22] curl_4.3.2          compiler_4.1.3      cli_3.2.0          
[25] rvest_1.0.2         htmlTable_2.4.0     xml2_1.3.3         
[28] labeling_0.4.2      entropy_1.3.1       sass_0.4.1         
[31] checkmate_2.1.0     digest_0.6.29       foreign_0.8-82     
[34] base64enc_0.1-3     jpeg_0.1-9          pkgconfig_2.0.3    
[37] htmltools_0.5.2     highr_0.9           dbplyr_2.1.1       
[40] fastmap_1.1.0       htmlwidgets_1.5.4   rlang_1.0.2        
[43] readxl_1.4.0        rstudioapi_0.13     farver_2.1.0       
[46] visNetwork_2.1.0    jquerylib_0.1.4     generics_0.1.2     
[49] jsonlite_1.8.0      crosstalk_1.2.0     vroom_1.5.7        
[52] Matrix_1.4-1        Rcpp_1.0.8.3        munsell_0.5.0      
[55] fansi_1.0.3         lifecycle_1.0.1     stringi_1.7.6      
[58] yaml_2.3.5          plyr_1.8.7          grid_4.1.3         
[61] parallel_4.1.3      crayon_1.5.1        haven_2.4.3        
[64] splines_4.1.3       pander_0.6.5        hms_1.1.1          
[67] pillar_1.7.0        reshape2_1.4.4      reprex_2.0.1       
[70] glue_1.6.2          evaluate_0.15       V8_4.1.0           
[73] latticeExtra_0.6-29 modelr_0.1.8        vctrs_0.4.0        
[76] png_0.1-7           tzdb_0.3.0          cellranger_1.1.0   
[79] gtable_0.3.0        assertthat_0.2.1    cachem_1.0.6       
[82] xfun_0.30           broom_0.7.12        memoise_2.0.1      
[85] cluster_2.1.2       ellipsis_0.3.2      ROCR_1.0-11        </code></pre>

</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="saved-data-objects.html"><span class="header-section-number">A</span> Saved Data Objects</a></div>
<div class="next"><a href="references.html">References</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#software-and-packages-used"><span class="header-section-number">B</span> Software and Packages Used</a></li>
<li><a class="nav-link" href="#basic-dependencies"><span class="header-section-number">B.1</span> Basic dependencies</a></li>
<li><a class="nav-link" href="#available-on-machine-used"><span class="header-section-number">B.2</span> Available on machine used</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
<li><a id="book-source" href="https://github.com/tmbuza/microbiome-part1/blob/main/99_appendix.Rmd">View source <i class="fas fa-air-freshener"></i></a></li>
          <li><a id="book-edit" href="https://github.com/tmbuza/microbiome-part1/edit/main/99_appendix.Rmd">Edit this page <i class="fas fa-air-freshener"></i></a></li>
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>Getting Started with Microbiome Data Analysis in R</strong>: Integrated solutions for better results" was written by Teresia Mrema-Buza, A Microbiome Computational Scientist and Owner of the Complex Data Insights, LLC, USA. It was last built on 2022-05-17.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer>
</body>
</html>
